src/MongoDB.Driver/Linq/MongoQueryable.cs

```diff
@@ -629,73 +630,58 @@ public static Task<int> CountAsync<TSource>(this IMongoQueryable<TSource> source
                 quotedPartitionByFields = Expression.Constant(null, typeof(Expression<Func<TSource, object>>[]));
             }
 
-            return (IMongoQueryable<TSource>)source.Provider.CreateQuery<TSource>(
+            return (IQueryable<TSource>)source.Provider.CreateQuery<TSource>(
                 Expression.Call(
                     GetMethodInfo(MongoQueryable.Densify, source, field, range, partitionByFields),
-                    Expression.Convert(source.Expression, typeof(IMongoQueryable<TSource>)),
+                    source.Expression,
                     Expression.Quote(field),
                     Expression.Constant(range),
                     quotedPartitionByFields));
         }
 
-        /// <summary>
-        /// Returns distinct elements from a sequence by using the default equality comparer to compare values.
-        /// </summary>
-        /// <typeparam name="TSource">The type of the elements of <paramref name="source" />.</typeparam>
-        /// <param name="source">The <see cref="IMongoQueryable{TSource}" /> to remove duplicates from.</param>
-        /// <returns>
-        /// An <see cref="IMongoQueryable{TSource}" /> that contains distinct elements from <paramref name="source" />.
-        /// </returns>
-        public static IMongoQueryable<TSource> Distinct<TSource>(this IMongoQueryable<TSource> source)
```

Comment:
Remove all extension methods that only mirror Queryable methods.

That still leaves a lot of custom extension methods!

Reply:
Still nice to remove the duplication of mirrored extension methods thus focusing on the custom ones.